callstack/react-native-paper
# refactor: reuse RadioButton context in RadioButton/Android/IOS

### Motivation
Resolve #1117
~~I couldn't think in any reasonable case that requires the use of either `RadioButton.Android` or `RadioButton.IOS` because `RadioButton` should do the job of rendering the right component.~~

~~If we are going to keep these two components exposed we'll need to get the right props from the context like `RadioButton` does when it is wrapped by a `RadioButton.Group`.~~

~~What do you guys think?~~

### Update
In order to reuse `RadioButton` [context implementation](https://github.com/callstack/react-native-paper/blob/master/src/components/RadioButton.tsx#L103), I created a new HOC [withRadioContext](https://github.com/callstack/react-native-paper/blob/d7cd76333abd0c002666fad4c95c9de08a7b14e9/src/components/RadioButton/WithRadioContext.tsx) to use in `RadioButtonAndroid` and `RadioButtonIOS`.

### Test plan
1. Download this branch
2. Check `Radio Button` and `Radio Button Group` sections on example app.

### Case 1 - Using `RadioButton.Android` and `RadioButton.IOS` without `RadioButton.Group`
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/6487206/61882158-4673c900-aece-11e9-8b0b-0420e0fa0237.gif)

### Case 2 - Using `RadioButton.Android` and `RadioButton.IOS` wrapped by `RadioButton.Group`
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/6487206/61882452-fa755400-aece-11e9-88b6-51752c2a3630.gif)
